### PR TITLE
Proxy and template literals

### PR DESCRIPTION
**Pull request by tkh44**
cc @FWeinb

### CODE REVIEW

**Comment by tkh44** [comment]
You are now a contributor to this project (if you want). It's your call when to merge this 👍 I'm going to put this in emotion-js org or something like that later.

**Comment by FWeinb** [comment]
Thanks for making me a contributor! I would like to discuss your intention with this library a little further. Some thoughts of mine:
1. If these changes are merged it would mean that `retarget` can only be used in environments where there is [proxy support](https://caniuse.com/#feat=proxy). As of now this means no IE support at all. To support IE a babel-plugin (or a babel-plugin-macro is needed) 
2. As it is implemented now one can mix-and-match proxy and tagged template syntax (See [these tests](https://github.com/FWeinb/retarget/blob/1dcfc559149647944546a5c87debe7f4019913fd/test/mixed.test.js)). Technically this is cool as hell but I don't think someone could possible want that. 
3. Performance: I would really like to get some benchmarks to compare `retarget` with e.g. [_.get](https://lodash.com/docs/4.17.5#get)
4. As talked about in 1. I think this whole concept could be implemented as a `babel-plugin-macro` with nearly zero runtime overhead for both the tagged template literal syntax and for the proxy syntax. 

Thanks for trusting me to merge this on my own terms. Looking forward to some feedback.

**Comment by tkh44** [comment]
I wouldn't mind the idea of shipping the library with the Proxy and using the macro as a way for developers to support ie. If the performance numbers show an improvement we can label that as another advantage.

**Comment by tkh44** [comment]
I will say this though. Each barrier to entry using this lib has a cost as I learned with https://emotion.sh. The challenge is finding a good balance.

**Comment by FWeinb** [comment]
Thanks for your feedback. In the current state this code won't run without proxy support. Even if the author would only use the tagged template syntax. That is because I am using a proxy as a default export (see [this line](https://github.com/FWeinb/retarget/blob/cf2b578077f8dd127cf5956f051b18c61b37b9b6/src/index.js#L104)). 

This could be solved with a feature detect of Proxy support and exporting the [`rootInstance`](https://github.com/FWeinb/retarget/blob/cf2b578077f8dd127cf5956f051b18c61b37b9b6/src/index.js#L91-L101) directly. As a consequence we should setup a browser test that will ensure the feature test is working correctly and gives the correct result.

To further emphasis on the separation between proxy and tagged template syntax I would remove the mix-and-match proxy support (mentioned in 2)) that would simplify the implementation too. 

I did some very basic performance tests and it looks like the proxy implementation is a bit faster that the tagged template one (tested in Chrome 64) but these tests are not very accurate in measuring the actual performance. 

In conclusion I think I will do the following:
- [x] Ensure that tagged template syntax is working without proxy support
- [x] Remove mix-and-match proxy/tagged template syntax support
